content/articles/angular-12-dependency-injection/index.md

```diff
@@ -0,0 +1,244 @@
+
+```bash
+ng g c student
+```
+
+Output:
+
+```
+CREATE src/app/student/student.component.css (0 bytes)
+CREATE src/app/student/student.component.html (26 bytes)
+CREATE src/app/student/student.component.spec.ts (635 bytes)
+CREATE src/app/student/student.component.ts (273 bytes)
+UPDATE src/app/app.module.ts (400 bytes)
+```
+Now proceed and edit the `src/app/student/student.component.ts` as shown below:
+
+```ts
+import { Component, OnInit } from '@angular/core';
+import {StudentListService} from '../student-list.service';
+import {Student} from '../student';
+
+@Component({
+  selector: 'app-student',
+  templateUrl: './student.component.html',
+  styleUrls: ['./student.component.css']
+})
+export class StudentComponent implements OnInit {
+
+// local array tohold the list of students
+  students: Student[];
+  // we're injecting the studentListService (this is Dependency injection in action)
+  constructor(private studentListService: StudentListService) { }
+
+  ngOnInit() {
+    this.getStudentsList();
+  }
+  //getting the list of students and assigning the list to students array
+  getStudentsList() {
+    this.students = this.studentListService.getInternsDetails();
+  }
+}
+
+```
+In the above component class, we've defined the `students` array that we will use to hold the list of students. We then inject the ``StudentListService` that we previously created. This is how Dependency injection is achieved in Angular.
+
+We've also created the `getStudentsList()` method that we're using to get student details from the service. We then assign the results to the students array we've locally created. We call this method in the `ngOnInit()` method to get students' details on page load.
+
+### Displaying data from a service
+Now that we've consumed our service let's proceed and display these data on the browser.
+
+Let's edit the `src/app/student/student.component.html` as shown below:
+
+```html
+<div>
+<table class="table table-stripped table-active">
+  <thead class="thead-light">
+  <th>#</th>
+  <th>First Name</th>
+  <th>Last Name</th>
+  <th>Admission Number</th>
+  <th>Course</th>
+  <th>Year Of Study</th>
+  <th>Reported On</th>
+  <th>College</th>
+  </thead>
+  <tbody>
+  <tr *ngFor="let student of students;let i=index">
+    <td>{{i+1}}</td>
+    <td>{{student.studentFirstName}}</td>
+    <td>{{student.studentLastName}}</td>
+    <td>{{student.studentRegistrationNumber}}</td>
+    <td>{{student.studentCourse}}</td>
+    <td>{{student.studentYearOfStudy}}</td>
+    <td>{{student.reportingDate}}</td>
+    <td>{{student.college}}</td>
+  </tr>
+  </tbody>
+</table>
+</div>
+
+```
+
+Now run your Angular application.
+
+![Student table](/engineering-education/angular-di/output.png)
+
+### Conclusion
+In this tutorial, we've discussed a wide range of concepts of Angular DI with examples. We've seen how the services are created and then consumed within a component. We've also gone a step further to test our application on the browser.
```

Comment:
Conclusion should cover what has been covered on the entire article. Kindly fix.